### PR TITLE
feat(orc8r): upgrade orchestrator docker base image

### LINKED ISSUE #14344: Dockerfiles in orc8r/cloud/docker are updated to a more recent ubuntu version
Follow-up from #14343

- https://github.com/nodesource/distributions

### PR DESCRIPTION
**Pull request by Neudrino**
## Summary

- Closes #14344
- Upgrade the orchestrator base image to **focal** LTS
- Upgrade nginx base image to latest minor

## Test Plan

- CI: https://github.com/magma/magma/actions/workflows/build_all.yml?query=branch%3Aorc8r%2Fdocker-base-image-version-upgrade

## Additional Information

- fluentd was already at the newest version in https://github.com/magma/magma/pull/12940
- An upgrade to `jammy` seems not easily possible, due to a very weird bug.
  - In the dp-workflow, one will get error when trying to install openjdk-8-jre-headless.
    ```
    # There is insufficient memory for the Java Runtime Environment to continue.
    # Cannot create GC thread. Out of system resources.
    ```
  - This error is not seen, when using focal.
  - This error is not seen, when building in build-all.
  - This error is not seen, when installing this package locally in and `ubuntu:jammy` docker image.
  - So something must be very brittle in the setup of dp-workflow together with the Github action runners.
  - Strange enough, directly before
    ```
    free -h
                  total        used        free      shared  buff/cache   available
    Mem:          6.8Gi       1.3Gi       363Mi        21Mi       5.1Gi       5.2Gi
    Swap:         4.0Gi       0.0Ki       4.0Gi
    ```

### CODE REVIEW

**Review by ShubhamTatvamasi — APPROVED** [review]

**Comment by sebathomas** [comment]
I just found this, maybe the JDK insufficient memory issue depends on the Docker version:
https://stackoverflow.com/questions/72841549/container-fails-to-start-insufficient-memory-for-the-java-runtime-environment-t

**Review by sebathomas — CHANGES_REQUESTED** [review]
Looking at the CI, it seems there is a timeout in the orc8r docker build with the changes here (or it becomes extremely slow, but then we definitely shouldn't merge it yet).

**Comment by Neudrino** [comment]
> I just found this, maybe the JDK insufficient memory issue depends on the Docker version: https://stackoverflow.com/questions/72841549/container-fails-to-start-insufficient-memory-for-the-java-runtime-environment-t

Interesting, but that makes upgrading to `jammy` clearly a no-opt for now.

**Comment by Neudrino** [comment]
> Looking at the CI, it seems there is a timeout in the orc8r docker build with the changes here (or it becomes extremely slow, but then we definitely shouldn't merge it yet).

:+1: Was waiting for the job result. Good that you noticed.
```
Setting up tzdata (2022f-0ubuntu0.20.04.1) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
Configuring tzdata
------------------

Please select the geographic area in which you live. Subsequent configuration
questions will narrow this down by presenting a list of cities, representing
the time zones in which they are located.

  1. Africa      4. Australia  7. Atlantic  10. Pacific  13. Etc
  2. America     5. Arctic     8. Europe    11. SystemV
  3. Antarctica  6. Asia       9. Indian    12. US
```
Have to deactivate interactivity here.

**Comment by sebathomas** [comment]
Looks like the minikube deployment in the DP job has some issues now.

**Comment by wolfseb** [comment]
When I run the newly built image for the orc8r controller with the changes from this PR, there is an error in the logs:
`/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found`
Is this of any concern? (this error does not appear on master)
Installing the library `libc6` does not seem to help either.

**Comment by wolfseb** [comment]
The production containers still uses xenial:

```
# ------------------------------------------------------------------------------
# Production
# ------------------------------------------------------------------------------
FROM ubuntu:xenial
```

**Comment by sebathomas** [comment]
I had a short look. It seems that Ubuntu Focal has a more recent version of supervisor (4.1 instead of 3.2) that is not compatible with the current setup. Installing `supervisor_logger.py` into python3.8 instead of python2.7 helps and the services will start but there is no logging. It seems that changes to the logger file and/or supervisor config are needed.

**Comment by Neudrino** [comment]
Let's see if smaller steps are easier: trying `bionic`.

File: orc8r/cloud/Makefile
Comment (sebathomas) [inline orc8r/cloud/Makefile:178]: It seems we can't use node 18 if we upgrade to bionic:
https://github.com/nodesource/distributions/issues/1392

**Comment by sebathomas** [comment]
The current version doesn't exactly reproduce the old logging yet (have to get the header of the event to find out which microservice logged something) but it seems to be working now.

File: orc8r/cloud/docker/controller/supervisor_logger.py
Comment (Neudrino) [inline orc8r/cloud/docker/controller/supervisor_logger.py]: Duplicated comment.

**Comment by sebathomas** [comment]
I finally got the logging format to work again. Looking at the output of `docker-compose logs -f controller`, it seems to be good again.

I would still postpone the merge a little bit until we are sure the new artifactory is not causing any issues.

File: orc8r/cloud/docker/controller/supervisor_logger.py
Comment (nstng) [inline orc8r/cloud/docker/controller/supervisor_logger.py:40]: Why is a constant value returned here? Do we expect that the loop is ended before if `2!=len(data)` or 'data!=OK'?

File: orc8r/cloud/docker/controller/supervisor_logger.py
Comment (nstng) [inline orc8r/cloud/docker/controller/supervisor_logger.py:43]: Should this react the same as before? I don't see This at a glance - can someone elaborate?

Comment (sebathomas) [reply orc8r/cloud/docker/controller/supervisor_logger.py:43]: This was the most complicated part of the change.

In the old version, the log message in `data` was written to stdout in the main loop and then processed in the result handler as part of the response. After the update, supervisor will just stop logging anything if you do that because it violates the event listener protocol, see for example: https://github.com/Supervisor/supervisor/issues/1417

In the new version, I implemented a result handler that takes the log message from the log events. I think the only way to see that it behaves the same way is to compare the controller logs before and after the change.

**Review by nstng — APPROVED** [review]

**Review by sebathomas — APPROVED** [review]
Okay, I think the artifactory switch is now sufficiently stable that I will merge it, even if it might break something in the orc8r.

**Pull request merged**
